src/application_logic/storage/createItem.ts

```diff
@@ -0,0 +1,47 @@
+import { showTodoArea } from "components/showTodos/showTodos";
+import { Timestamp, addDoc } from "firebase/firestore";
+import { auth } from "../auth";
+import { ProjectData, TodoData, Project, Todo } from "types";
+import { projects, todosOfProject } from "./useDb";
+
+// factory for projects
+const createProjectData = (name: string, isInbox?: boolean): ProjectData => {
```

Comment:
What I find a bit confusing in this file, is that you mix concepts. You have two factory methods (createProject/TodoData) and one api call (createItem). I think it is confusing that they live in the same module. If I'd see createItem first, I'd expect createProject/TodoData to be API calls too.

That's something I would change.Either the factories are only used nternally (by createItem) or they should move somewhere else. That you export createProjectData, suggests it is used somewhere else. Maybe a factory.ts would be a good first step for you!?